js/components/Signup.jsx

```diff
@@ -15,6 +15,75 @@ var Signup = React.createClass({
            };
   },
 
+  componentDidMount: function() {
+    // Configure to work with localhost or repcoin.net
+    if (document.domain === 'localhost') {
+      var appId = '898342783518783';
+    } else {
+      var appId = '894010190618709';
+    }
+    window.fbAsyncInit = function() {
+      FB.init({
+        appId      : appId,
+        cookie     : true,  // enable cookies to allow the server to access
+                            // the session
+        xfbml      : true,  // parse social plugins on this page
+        version    : 'v2.1' // use version 2.1
+      });
+    };
+
+    // Load the SDK asynchronously
+    (function(d, s, id) {
+      var js, fjs = d.getElementsByTagName(s)[0];
+      if (d.getElementById(id)) return;
+      js = d.createElement(s); js.id = id;
+      js.src = "//connect.facebook.net/en_US/sdk.js";
+      fjs.parentNode.insertBefore(js, fjs);
+    }(document, 'script', 'facebook-jssdk'));
+  },
+
+  handleFacebookClick: function(e) {
+    e.preventDefault();
+    FB.getLoginStatus(this.statusCallback);
+  },
+
+  statusCallback: function(response) {
+    if (response.status === 'connected') {
+      this.loginUser(response.authResponse.accessToken);
+    } else {
+      FB.login(this.loginCallback, true);
```

Comment:
If the user is already logged in to facebook, just log them into the app. Otherwise, log them into facebook